### ISSUE TITLE
Support for vectorized environments? 

### ISSUE THREAD

**Issue by Sohojoe** [tensorflow/agents#17]
Do you support vectorized environments? 

I am the author of [MaratonEnvs](https://github.com/Unity-Technologies/marathon-envs), a set of continuous control environments built in Unity+ml-agents. I use vectorized environments to scale the number of duplicate agents within the environment which gives a significant performance boost over MPI style strategies. It would be nice to support tf-agents

[VecEnv base class from openai.baselines](https://github.com/openai/baselines/blob/b875fb7b5e4feb85b9f1f1bf4e78f64c75595664/baselines/common/vec_env/vec_env.py)

**Comment by jrabary** [comment]
As far as I know, there is already a process based environment in https://github.com/tensorflow/agents/blob/master/tf_agents/environments/parallel_py_environment.py.
I think it is not using MPI but can be used as a starting point.

**Comment by Sohojoe** [comment]
@jrabary thanks - I will check it out. I think this is what I need as I have multiple instances of the same environment within the environment (if that makes sense!)

**Comment by oars** [comment]
Hey @Sohojoe,

@jrabary pointed out the parallel_py_environment which I'm not sure is the right solution here. Parallel envs are aimed at starting multiple instances of an environment. I'm assuming in your case you have multiple arenas in the same unity instance though.

In that scenario I believe the [batched_py_environment](https://github.com/tensorflow/agents/blob/master/tf_agents/environments/batched_py_environment.py) will be a better fit. I only took a quick look but it seems fairly similar to the VecEnv you reference. 

We have some preliminary work on supporting Unity envs. I could easily reproduce the way the unity wrapper for Gym works, but I found the way brains and rendering was not as general as I would have liked. It's probably a good starting point though.
